### PR TITLE
Update docval to accept fully qualified class names

### PR DESCRIPTION
**Pull request by rly**
## Motivation

Follow-up to https://github.com/hdmf-dev/hdmf/pull/1031 to allow a "type" argument in docval that is the name of a class to be linked to its documentation.

## How to test the behavior?

Use this syntax for `'type'` if you want to create a proper doc link to the `LinkSpec` class:
```python
@docval({'name': 'spec', 'type': 'hdmf.spec.spec.LinkSpec', 'doc': 'the specification for the object to link to'})
    def set_link(self, **kwargs):
        pass
```

## Checklist

- [x] Did you update `CHANGELOG.md` with your changes?
- [x] Does the PR clearly describe the problem and the solution?
- [x] Have you reviewed our [Contributing Guide](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst)?
- [x] Does the PR use "Fix #XXX" notation to tell GitHub to close the relevant issue numbered XXX when the PR is merged?

### CODE REVIEW

File: tests/unit/test_io_hdf5_streaming.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class TestRos3(TestCase):
     def setUp(self):
         # Skip ROS3 tests if internet is not available or the ROS3 driver is not installed
         try:
+            # this is a 174 KB file
```
Comment (stephprince) [inline tests/unit/test_io_hdf5_streaming.py:22]: not sure if this was meant to be added here?

**Review by stephprince — COMMENTED** [review]
I think one more type that could be changed with this new docval behavior is `'type': 'HERD'` to `'type':'hdmf.common.resources.HERD'`? It would apply [here](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/backends/hdf5/h5tools.py#L366), [here](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/backends/io.py#L78), and [here](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/backends/io.py#L78)

otherwise looks good to me!

**Comment by rly** [comment]
> I think one more type that could be changed with this new docval behavior is `'type': 'HERD'` to `'type':'hdmf.common.resources.HERD'`? It would apply [here](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/backends/hdf5/h5tools.py?rgh-link-date=2024-01-24T01%3A03%3A01Z#L366), [here](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/backends/io.py?rgh-link-date=2024-01-24T01%3A03%3A01Z#L78), and [here](https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/backends/io.py?rgh-link-date=2024-01-24T01%3A03%3A01Z#L78)

I am not sure how I missed that. Good catch!

File: tests/unit/test_io_hdf5_streaming.py (hunk 1 above)
Comment (rly) [reply tests/unit/test_io_hdf5_streaming.py:22]: oops. i did not mean to add that. it is technically unrelated but I think it is a useful comment and it's kind of silly to make it its own PR, so i would prefer ot keep it

**Review by stephprince — APPROVED** [review]

**Pull request merged**